src/app/core/services/shelf-configuration-state.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core'
+import { BehaviorSubject, Observable } from 'rxjs'
+import { ConnectionStatusObserver } from './connection-status-observer.service'
+import { EventSubscription } from '../../event-system/abstractions/event-observer.service'
+import { MatSnackBar } from '@angular/material/snack-bar'
+import { ConfigurationEventObserver } from './configuration-event-observer'
+import { ShelfConfigurationUpdatedEvent } from '../models/configuration-event'
+import { ShelfActionPanelConfiguration, ShelfConfiguration } from 'src/app/shared/models/shelf-configuration'
+import { ConfigurationService } from 'src/app/shared/services/configuration.service'
+
+@Injectable()
+export class ShelfConfigurationStateService {
```

Comment:
I use it because of the data centralization logic and I don't need to resync the data after adding/importing (or provide the data from child to parent component) or other operations. I consider it a good practice and a fairly familiar practice by all developers in the project. In addition, I do not know where it is expected to be used and how exactly the information is stored here.

Reply:
StateServices is a practice to be used when we have data that needs to go across multiple components/services that or to prevent duplicating complex logic maintaining state.

This stateService does neither.

What this stateService currently does falls under the responsibility of the "mother/smart" component in a "mother/child" (or smart/dumb) component hierarchy which is what we want in ngSofie.

This stateService currently only fetches the ShelfConfigurations and nothing else and it is only used by _a single place_ in the code (the `action-panel.component.ts`). The ActionPanelComponent should simply inject the `ConfigurationService` directly itself instead of the stateService.

Introducing a stateService introduces an extra layer of complexity. We need to have good arguments for introducing complexity. The arguments: "We do this elsewhere in the code" or "People know how to use this" are not good enough arguments.